add hostname lookup

twouters opened the pull request:
I've hacked in an extra hostname header.
Not sure if it's useful for you and/or done the right way. Comments welcome :-)

I haven't been able to test with punycode and multiple PTR records yet.

I'm also not very happy with the changes in apt_test.go, but I don't know another way to predict the correct hostname.

martinp commented:
I can see why this might be useful, but at the current request rate it would cause too many lookups. Thanks though.

martinp commented:
I suppose I could add it as an optional feature, controlled with a flag. What do you think?

twouters commented:
That would be great, if you want I can update the pull request

martinp commented:
Sure. Add a flag for enabling this feature, that defaults to false.

twouters commented:
I've added a [-r|--reverse] flag - I first opted for [-h|--hostname] but I didn't want to use -h flag.
I also removed the hostname lookup from the tests.

martinp commented on api/api.go:
```diff
@@ -30,6 +31,7 @@ var cliUserAgentExp = regexp.MustCompile("^(?i)(curl|wget|fetch\\slibfetch)\\/.*
 type API struct {
 	db       *geoip2.Reader
 	CORS     bool
+	Reverse  bool
```
Can you change this to `ReverseLookup`? Same for the field in `opts`.

twouters commented:
Changed Reverse to ReverseLookup

martinp commented:
Thanks. Can you also run gofmt on the changed files?

twouters commented:
reformatted as requested

martinp commented:
Thanks!
